lib/backoff-with-jitter.js

```diff
@@ -0,0 +1,28 @@
+'use strict'
+
+/**
+ * This implements exponential backoff with "decorrelated jitter" for use in
+ * failing HTTP calls and their retries.  Although the HTTP calls shouldn't be in
+ * contention with other clients, the jitter will help alleviate a flood
+ * of connections to the server in the event LogDNA suddenly comes back
+ * online after being unavailable.
+ *
+ * @see https://aws.amazon.com/blogs/architecture/exponential-backoff-and-jitter/
+ *
+ * @param   {Number} base The base sleep time to be used
+ * @param   {Number} cap The maximum sleep time allowable in milliseconds
+ * @param   {Number} lastSleep The last (or starting) sleep time used
+ * @returns {Number} sleep time in seconds
```

Comment:
`calculated sleep time` since unit of measure is arbitrary and based on passed-in values.